### PR TITLE
feat: Add server-side rendering for MDX content

### PR DESCRIPTION
**Pull request by franckgaudin**

### CODE REVIEW

File: apps/docs/app/components/[slug]/page.tsx [hunk 1]
```diff
@@ -1,45 +1,76 @@
...
+                <section>
+                    <h2>Accessibility</h2>
+                </section>
+
+                <section>
+                    <h2>Related</h2>
+                </section>
+
+                <div>
+                    <button type="submit">Preview</button>
```
Comment (alexasselin008) [inline apps/docs/app/components/[slug]/page.tsx:71]: i know this is a test, but there is 2 submit button, that's a bit awkward

**Review by alexasselin008 — APPROVED** [review]

File: apps/docs/app/components/[slug]/page.tsx (hunk 1 above)
Comment (franckgaudin) [reply apps/docs/app/components/[slug]/page.tsx:71]: indeed it's just to place them, they will be integrated in a more formal way in another PR

File: apps/docs/content/components/button.mdx [hunk 2]
```diff
@@ -4,4 +4,9 @@ description: The button Component
 status: "Released"
 ---
 
-# Button
+## Usage
+
+```jsx
```
Comment (fraincs) [inline apps/docs/content/components/button.mdx:9]: rehype-pretty-code is not running here. Something to look into as currently other mdx files are parsed and 
gets an better experience, syntax highlighting as well as a copyCode button.

 image
image

Comment (franckgaudin) [reply apps/docs/content/components/button.mdx]: Done

File: apps/docs/utils/rehypeConfig.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
+import {visit} from "unist-util-visit";
+import rehypePrettyCode from "rehype-pretty-code";
+
+export const rehypeOptions = {
```
Comment (alexasselin008) [inline apps/docs/utils/rehypeConfig.js:4]: Can this file be a .ts or use jsdoc for some type safety?

**Review by fraincs — DISMISSED** [review]

File: apps/docs/utils/getComponentDetails.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import path from "path";
+import fs from "fs";
+import { compileMDX } from "next-mdx-remote/rsc";
+import { components } from "@/components/ui/mdx/components.tsx";
+import { rehypePluginOptions } from "@/utils/rehypeConfig";
+
+async function parseFrontMatter(fileContent: string) {
+    const { content, frontmatter } = await compileMDX<{ title: string; description: string; status: string }>({
+        source: fileContent,
+        options: { parseFrontmatter: true, mdxOptions: { remarkPlugins: [], rehypePlugins: rehypePluginOptions as unknown as any[]}},
```
Comment (vicky-comeau) [inline apps/docs/utils/getComponentDetails.ts:10]: I'm guessing the linting error you're getting is from this any

**Review by alexasselin008 — APPROVED** [review]

**Pull request merged**